pkg/virt-controller/services/template.go

```diff
@@ -1547,44 +1546,64 @@ func getIfaceByName(vmi *v1.VirtualMachineInstance, name string) *v1.Interface {
 	return nil
 }
 
-func getCniAnnotations(vmi *v1.VirtualMachineInstance) (cniAnnotations map[string]string, err error) {
-	ifaceListMap := make([]map[string]string, 0)
-	cniAnnotations = make(map[string]string, 0)
+func getMultusCNIAnnotation(vmi *v1.VirtualMachineInstance) (string, error) {
+	var ifaceListMap []map[string]string
+
+	relevantNetworks := filterMultusNonDefaultNetworks(vmi)
+	for i, network := range relevantNetworks {
+		iface := getIfaceByName(vmi, network.Name)
+		namespace, networkName := getNamespaceAndNetworkName(vmi, network.Multus.NetworkName)
+		serializedInterface := mapifyMultusInterface(iface, networkName, namespace, i)
+		ifaceListMap = append(ifaceListMap, serializedInterface)
+	}
+
+	if len(ifaceListMap) > 0 {
+		ifaceJsonString, err := serializeMultusInterfaces(ifaceListMap)
+		if err != nil {
+			return "", err
+		}
+		return ifaceJsonString, nil
+	}
+	return "", nil
+}
+
+func serializeMultusInterfaces(ifaceListMap []map[string]string) (string, error) {
+	ifaceJsonString, err := json.Marshal(ifaceListMap)
+	if err != nil {
+		return "", fmt.Errorf("Failed to create JSON list from CNI interface map %s", ifaceListMap)
+	}
+	return string(ifaceJsonString), nil
+}
 
-	next_idx := 0
+func mapifyMultusInterface(multusInterface *v1.Interface, networkName string, namespace string, interfaceSuffix int) map[string]string {
```

Comment:
Sorry, but I lost track here.
- I do not understand the name of the function.
- Why there is no representation through a struct that is annotated. Having a proper name with ref to who/what defined it.
- Why do we need both `Interface` and `networkName`? The first already includes that interface/network name.